xmtp_mls/src/groups/mod.rs

```diff
@@ -415,26 +413,20 @@ where
 
         let provider = context.mls_provider();
         let mls_group = if let Some(group_id) = group_id {
-            OpenMlsGroup::new_with_group_id(
+            // TODO: For groups restored from backup, in order to support queries on metadata such as
+            // the group title and description, a stubbed OpenMLS group is created, and later overwritten
+            // when a welcome is received.
+            // To avoid potentially operating on this encryption state elsewhere, it may instead be better
+            // to store this metadata on the StoredGroup instead, and modify group metadata queries to also
+            // check the StoredGroup.
+            OpenMlsGroup::stub_from_backup_logged(
                 &provider,
-                &context.identity.installation_keys,
+                &context.identity,
                 &group_config,
                 GroupId::from_slice(group_id),
-                CredentialWithKey {
-                    credential: context.identity.credential(),
-                    signature_key: context.identity.installation_keys.public_slice().into(),
-                },
             )?
```

Comment:
@mchenani @neekolas This is the issue we found earlier, WDYT?

Reply:
using StoredGroup to enable metadata queries when restoring from backup instead of stubbed OpenMls makes sense to me 👍